packages/hint-doctype/README.md

```diff
@@ -0,0 +1,114 @@
+# Check that the DOCTYPE tag of the document is valid
+
+This hint checks if the HTML is using the most modern DOCTYPE.
+
+## Why is this important?
+
+> In HTML, the doctype is the required "<!DOCTYPE html>" preamble
+found at the top of all documents. Its sole purpose is to prevent
+a browser from switching into so-called “quirks mode” when rendering
+a document; that is, the "<!DOCTYPE html>" doctype ensures that the
+browser makes a best-effort attempt at following the relevant specifications,
+rather than using a different rendering mode that is incompatible
+with some specifications.
+
+***From [MDN glossary for DOCTYPE specification][docmdn].***
+
+## What does the hint check?
+
+This hint checks if the HTML is using the most modern and valid DOCTYPE.
+
+It checks that the DOCTYPE is in the first line and that there are no other
+lines before the DOCTYPE. This is important as some browsers,
+including versions of IE prior to IE10, trigger quirks mode if a comment
+occurs before the DOCTYPE.
+
+It checks that there is no additional info apart from the DOCTYPE on the same
+line and that it is not duplicated elsewhere in the document. To support older
+HTML content generators, it also accepts the legacy-compat DOCTYPE. Examples:
+
+`<!DOCTYPE html>`
+
+`<!doctype html>`
+
+`<!DOCTYPE html SYSTEM "about:legacy-compat">`
```

Comment:
I personally don't think we should recommend this. This is mostly for things such as [older outdated tools](http://bugzilla.bluegriffon.org/show_bug.cgi?id=634#c0) to not complain, not for browsers.

Also from: https://dev.w3.org/html5/html-author/#doctype-declaration

> Note: The term "legacy-compat" refers to compatibility with legacy producers only. In particular, **it does not refer to compatibility with legacy browsers, which, in practice, ignore SYSTEM identifiers and DTDs.**

Thoughts? 

---

See also: https://hsivonen.fi/doctype/#handling

Reply:
@molant what do you think? That was a feedback added recently, but @alrra seems right.

Reply:
Cc @webhintio/contributors

Reply:
@CKGrafico @isabellachen Can you also change the code for this? Thanks!

Reply:
Yes, today we are going to fix this. Thanks!

Reply:
The changes have been made, docs updated :)